### PR TITLE
Adding in parameters for using a proxy host connection

### PR DESCRIPTION
**Pull request by pauldavidchristian**

### CODE REVIEW

**Comment by vamshi-sfdc** [comment]
- validation logic to move to validator class
- increment version in pom.xml

**Comment by pauldavidchristian** [comment]
@vamshi-sfdc Do you recall the name of the validator class? It doesn't jump out at me?

**Comment by adarsh-ramakrishna-sfdc** [comment]
@pauldavidchristian  : Validator classes for the arguments are in the package com.sforce.cd.apexUnit.arguments. Please follow PercentageInputValidator class; you need to implement IParameterValidator for the JCommander framework to execute the validation method for a given command line argument

**Comment by adarsh-ramakrishna-sfdc** [comment]
@pauldavidchristian : After looking at your validation code, I feel you can use PositiveIntegerValidator validator class . Please use the "validateWith" option in the parameter definition and mention the validator class

**Comment by pauldavidchristian** [comment]
@adarsh-ramakrishna-sfdc Changes are in. Please let me know if you see anything else.

File: src/main/java/com/sforce/cd/apexUnit/client/connection/ConnectionHandler.java [hunk 1]
```diff
@@ -110,6 +110,14 @@ private PartnerConnection createConnection() {
 			config.setUsername(CommandLineArguments.getUsername());
 			config.setPassword(CommandLineArguments.getPassword());
 			config.setAuthEndpoint(CommandLineArguments.getOrgUrl() + "/services/Soap/u/" + SUPPORTED_VERSION);
+			
+			if (CommandLineArguments.getProxyHost() != null &&  CommandLineArguments.getProxyPort() !=null ){
+				LOG.info("Setting proxy configuraiton to " +
```
Comment (adarsh-ramakrishna-sfdc) [inline src/main/java/com/sforce/cd/apexUnit/client/connection/ConnectionHandler.java]: Not a major change.. But Can this be a debug log ?

File: src/main/java/com/sforce/cd/apexUnit/client/connection/ConnectionHandler.java [hunk 2]
```diff
@@ -169,7 +177,13 @@ private BulkConnection createBulkConnection() {
 		config.setRestEndpoint(restEndPoint);
 		config.setCompression(true);
 		config.setTraceMessage(false);
-		// config.setProxy("Proxy", 8080);
+
+		if (CommandLineArguments.getProxyHost() != null &&  CommandLineArguments.getProxyPort() !=null ){
+			LOG.info("Setting proxy configuraiton to " +
```
Comment (adarsh-ramakrishna-sfdc) [inline src/main/java/com/sforce/cd/apexUnit/client/connection/ConnectionHandler.java]: Not a major change.. But Can this be a debug log ?

File: README.md [hunk 3]
```diff
@@ -73,6 +73,8 @@ Optional Parameters:
 - -manifest.files.with.test.class.names.to.execute : Manifest files containing the list of test classes to be executed
 - -manifest.files.with.source.class.names.for.code.coverage.computation : Manifest files containing the list of Apex classes for which code coverage is to be computed
 - -max.test.execution.time.threshold : Maximum execution time(in minutes) for a test before it gets aborted
+- -proxy.host : Proxy host for external access
+- -proxy.port : Proxy port for external access
```
Comment (adarsh-ramakrishna-sfdc) [inline README.md]: Can you please update the sample command with the new CLI options  in the README file as well?

Comment (adarsh-ramakrishna-sfdc) [inline README.md]: And also in the command:

Comment (pauldavidchristian) [inline README.md]: Done.

Comment (adarsh-ramakrishna-sfdc) [inline README.md]: Thanks @pauldavidchristian

**Pull request merged**